### PR TITLE
Add HTTPS support to standard authentication service

### LINKED ISSUE #464: Add HTTPS support for standard authentication service

### PR DESCRIPTION
**Pull request by lulf**
* Add keystore and import certs from secure volume
* Modify configuration to load keystore for HTTPS endpoint
* Modify keycloak route and services to use HTTPS
* Fetch certificate and use in keycloak-controller
* Trust all certs for systemtests keycloak client

Fixes #464

### CODE REVIEW

File: keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakParams.java [hunk 1]
```diff
@@ -15,28 +15,61 @@
...
+            JcaX509CertificateConverter converter = new JcaX509CertificateConverter();
+            Certificate certificate = converter.getCertificate((X509CertificateHolder) pemParser.readObject());
+            keyStore.setCertificateEntry("standard-authservice", certificate);
+
+            return keyStore;
+        } catch (Exception ignored) {
+            log.warn("Error creating keystore for authservice CA", ignored);
+            throw ignored;
+        }
     }
```
Comment (rgodfrey) [inline keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakParams.java]: Do we actually need to use bouncycastle here, for instance https://github.com/apache/qpid-broker-j/blob/master/broker-core/src/main/java/org/apache/qpid/server/transport/network/security/ssl/SSLUtil.java#L310-L330 reads X509 certificates from PEM files without needing bouncycastle

File: agent/www/help.html [hunk 2]
```diff
@@ -1025,11 +1025,11 @@ <h5 id="amqp_net_lite">AMQP.Net Lite</h5>
...
+Last updated 2017-11-15 09:05:22 CET
 </div>
 </div>
 <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
 <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
 <script>hljs.initHighlighting()</script>
 </body>
-</html>
\ No newline at end of file
+</html>
```
Comment (rgodfrey) [inline agent/www/help.html]: Not sure it was really worth commit this change :)

File: Jenkinsfile [hunk 3]
```diff
@@ -34,7 +34,7 @@ node('enmasse') {
             stage('system tests') {
                 withCredentials([string(credentialsId: 'openshift-host', variable: 'OPENSHIFT_URL'), usernamePassword(credentialsId: 'openshift-credentials', passwordVariable: 'OPENSHIFT_PASSWD', usernameVariable: 'OPENSHIFT_USER')]) {
                     try {
-                        sh 'ARTIFACTS_DIR=artifacts OPENSHIFT_PROJECT=$BUILD_TAG ./systemtests/scripts/run_test_component.sh templates/install /var/lib/origin/openshift.local.config/master/admin.kubeconfig systemtests'
+                        sh 'ARTIFACTS_DIR=artifacts OPENSHIFT_PROJECT=${JOB_NAME::16}${BUILD_NUMBER} ./systemtests/scripts/run_test_component.sh templates/install /var/lib/origin/openshift.local.config/master/admin.kubeconfig systemtests'
                     } finally {
```
[the remark this answers is not in the document]
Comment (lulf) [reply Jenkinsfile]: No, but its an improvement to make jenkins build long branch names which it was having issues with.

File: keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakParams.java (hunk 1 above)
Comment (lulf) [reply keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakParams.java]: Didn't know about that, thanks!

**Pull request merged**